internal/pkg/scaffold/olm-catalog/descriptor.go

```diff
@@ -0,0 +1,544 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package catalog
+
+import (
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/operator-framework/operator-sdk/internal/annotations"
+	"github.com/operator-framework/operator-sdk/internal/pkg/scaffold"
+	"github.com/operator-framework/operator-sdk/internal/util/k8sutil"
+	"github.com/operator-framework/operator-sdk/internal/util/projutil"
+
+	olmapiv1alpha1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/gengo/parser"
+	"k8s.io/gengo/types"
+)
+
+const csvgenPrefix = annotations.SDKPrefix + ":gen-csv:"
+
+// setCRDDescriptorForGVK parses type and struct field declaration comments on
+// API types to populate a csv's spec.customresourcedefinitions.owned fields
+// for a given API identified by Group, Version, and Kind.
+// TODO(estroz): support ActionDescriptors parsing/setting.
+func setCRDDescriptorForGVK(crdDesc *olmapiv1alpha1.CRDDescription, gvk schema.GroupVersionKind) error {
+	if strings.Contains(gvk.Group, ".") {
+		gvk.Group = strings.Split(gvk.Group, ".")[0]
+	}
+	apisDir := filepath.Join(scaffold.ApisDir, gvk.Group, gvk.Version)
+	if _, err := os.Stat(apisDir); err != nil {
+		if os.IsNotExist(err) {
+			log.Infof(`API "%s" does not exist. Skipping CSV annotation parsing for this API.`, gvk)
```

Comment:
A little more specific on what wasn't found. Since it's a user message.
```suggestion
			log.Infof(`API directory "%s" does not exist. Skipping CSV annotation parsing for "%s" API.`, apisDir, gvk)
```